webapp/src/components/post_type_chat.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+import {useSelector} from 'react-redux';
+
+import styled from 'styled-components';
+
+import IconAI from 'src/components/ai_icon';
+
+const useAIAvailable = () => {
+    //@ts-ignore plugins state is a thing
+    return useSelector((state) => Boolean(state.plugins?.plugins?.[aiPluginID]));
+};
+
+const aiPluginID = 'mattermost-ai';
+
+const useCallsPostButtonClicked = () => {
+    return useSelector((state) => {
+        //@ts-ignore plugins state is a thing
+        return state['plugins-' + aiPluginID]?.callsPostButtonClickedTranscription;
```

Comment:
I don't want to translate this to typescript because it would be inconsistent with the rest of the code base, actually this is copied from calls plugin (that was typescription) and I changed it to javascript. I don't want to introduce more technology changes in this PR.

About using a different approach, I'm OK with that, but not as part of this PR. This PR only replicates an existing approach that we now works. We can decide to improve the approach but it would need coordination between the AI plugin, the Calls plugin and this plugin. That is a different effort in my opinion.

Reply:
> I don't want to translate this to typescript because it would be inconsistent with the rest of the code base, actually this is copied from calls plugin (that was typescription) and I changed it to javascript. I don't want to introduce more technology changes in this PR.

There is support for typescript in this codebase though I think https://github.com/mattermost/mattermost-plugin-zoom/blob/master/webapp/src/websocket/index.ts. Even though most of the codebase is js, I think we should use ts for new code if there is not a large barrier to introducing it

Reply:
Migrated to typescript